packages/block-library/src/template-part/index.php

```diff
@@ -65,16 +66,24 @@ function render_block_core_template_part( $attributes ) {
 		} else {
 			// Else, if the template part was provided by the active theme,
 			// render the corresponding file content.
-			$parent_theme_folders        = get_block_theme_folders( get_template() );
-			$child_theme_folders         = get_block_theme_folders( get_stylesheet() );
-			$child_theme_part_file_path  = get_theme_file_path( '/' . $child_theme_folders['wp_template_part'] . '/' . $attributes['slug'] . '.html' );
-			$parent_theme_part_file_path = get_theme_file_path( '/' . $parent_theme_folders['wp_template_part'] . '/' . $attributes['slug'] . '.html' );
-			$template_part_file_path     = 0 === validate_file( $attributes['slug'] ) && file_exists( $child_theme_part_file_path ) ? $child_theme_part_file_path : $parent_theme_part_file_path;
-			if ( 0 === validate_file( $attributes['slug'] ) && file_exists( $template_part_file_path ) ) {
-				$content = file_get_contents( $template_part_file_path );
-				$content = is_string( $content ) && '' !== $content
+			if ( 0 === validate_file( $attributes['slug'] ) ) {
+				$child_theme_folders     = get_block_theme_folders( $stylesheet );
+				$template_part_file_path = get_theme_file_path( '/' . $child_theme_folders['wp_template_part'] . '/' . $attributes['slug'] . '.html' );
+				$template                = get_template();
+
+				if ( $template !== $stylesheet ) {
+					$parent_theme_folders        = get_block_theme_folders( $template );
+					$parent_theme_part_file_path = get_theme_file_path( '/' . $parent_theme_folders['wp_template_part'] . '/' . $attributes['slug'] . '.html' );
+					if ( file_exists( $parent_theme_part_file_path ) ) {
+						$template_part_file_path = $parent_theme_part_file_path;
+					}
+				}
+				if ( file_exists( $template_part_file_path ) ) {
+					$content = file_get_contents( $template_part_file_path );
+					$content = is_string( $content ) && '' !== $content
```

Comment:
`file_get_contents()` will return a string, or `false` on failure. So perhaps we could replace the `is_string` condition with a check for the `false` scenario?
```suggestion
					$content = file_get_contents( $template_part_file_path );
					$content = false !== $content && '' !== $content
```

It's extremely unlikely that it returns `false` since we already know that the file exists from the above check... So we can even remove it completely without any significant risk. I think it would only fail in case there's a filesystem error, corrupt files etc 🤔

Reply:
```suggestion
					$content = file_get_contents( $template_part_file_path );
					$content = ! empty( $content ) 
```
How about something simple like this? @aristath